test/c_abi/main.zig

```diff
@@ -831,3 +832,159 @@ test "C ABI pointer sized float struct" {
     try expect(x.x == 3);
     try expect(x.y == 4);
 }
+
+//=== Helpers for struct test ===//
+pub inline fn expectSize(comptime t: type, size: u32) !void {
+    try expectEqual(size, @sizeOf(t));
+}
+
+pub inline fn expectAlign(comptime t: type, alignment: u32) !void {
+    try expectEqual(alignment, @alignOf(t));
+}
+
+pub fn expectFieldOffset(ps: anytype, pf: anytype, ofst: usize) !void {
+    try std.testing.expect(@ptrToInt(ps) <= @ptrToInt(pf));
+    try expectEqual(ofst, @ptrToInt(pf) - @ptrToInt(ps));
+}
+
+pub inline fn expectOk(c_err: c_int) !void {
+    if (c_err != 0) {
+        std.debug.print("ABI mismatch on field v{d}.\n", .{c_err});
+        return error.TestExpectedEqual;
+    }
+}
+
+pub inline fn expectFailX86(c_err: c_int) !void {
+    if (comptime !builtin.target.cpu.arch.isX86()) return expectOk(c_err);
+
```

Comment:
Not a big fan of this, would prefer the usual `if (comptime builtin.cpu.arch.isX86()) return error.SkipZigTest;`

Reply:
The idea is to avoid having skipped but valid tests. But as you want, since you're the person most likely to have to deal with that.

Reply:
What do you mean by valid?

Reply:
At some point someone might fix the C ABI issue. If they don't know about this test, they'll leave it as is. With my approach the test will fail and the Dev will be made aware of the test and quickly convert them to 'expectOk'.

With yours the test will continue to be skipped and the C ABI might break silently in the future.

Reply:
Someone working on the C ABI really should be checking if there are any tests being skipped just as they should do when working on any other tests.

Reply:
done